main.rb

```diff
@@ -1,27 +1,92 @@
-require_relative 'person'
-require_relative 'decorator'
-require_relative 'classroom'
-require_relative 'student'
-require_relative 'book'
-require_relative 'rental'
-
-person = Person.new(22, name: 'maximilianus')
-puts person.correct_name
-
-capitalized_person = CapitalizeDecorator.new(person)
-puts capitalized_person.correct_name
-
-capitalized_trimmed_person = TrimmerDecorator.new(capitalized_person)
-puts capitalized_trimmed_person.correct_name
-
-classroom = Classroom.new('Class A')
-student = Student.new(classroom, 'John Doe')
-book = Book.new('Ruby', 'Chris')
-rental = Rental.new('2023-06-30', book, person)
-
-puts "Classroom label: #{classroom.label}"
-puts "Student name: #{student.name}"
-puts "Book title: #{book.title} by #{book.author}"
-puts "Rental date: #{rental.date}"
-puts "Person name: #{person.name}"
-puts "Person's rentals: #{person.rentals.map { |r| r.book.title }}"
+require_relative 'app'
+
+def main_menu(_app)
+  puts "\nWelcome to the Library Management System!"
+  puts 'Please choose an option:'
+  puts '1. List all books'
+  puts '2. List all people'
+  puts '3. Create a person'
+  puts '4. Create a book'
+  puts '5. Create a rental'
+  puts '6. List rentals for a person'
+  puts '7. Quit'
+  print 'Enter your choice: '
+end
+
+def handle_choice(app, choice)
+  actions = {
+    1 => :handle_list_all_books,
+    2 => :handle_list_all_people,
+    3 => :handle_create_person,
+    4 => :handle_create_book,
+    5 => :handle_create_rental,
+    6 => :handle_list_rentals,
+    7 => :exit_app
+  }
+
+  action = actions[choice]
+
+  if action
+    send(action, app)
+  else
+    puts 'Invalid choice. Please try again.'
+  end
+end
+
+def handle_list_all_books(app)
+  app.list_all_books
+end
+
+def handle_list_all_people(app)
+  app.list_all_people
+end
+
+def handle_create_person(app)
+  print 'Enter person name: '
+  name = gets.chomp
+  print 'Enter person age: '
+  age = gets.chomp.to_i
+  print 'Enter person role (student/teacher): '
+  role = gets.chomp.downcase
+  app.create_person(name, age, role)
+end
+
+def handle_create_book(app)
+  print 'Enter book title: '
+  title = gets.chomp
+  print 'Enter book author: '
+  author = gets.chomp
+  app.create_book(title, author)
+end
+
+def handle_create_rental(app)
+  print 'Enter person ID: '
+  person_id = gets.chomp.to_i
+  print 'Enter book title: '
+  book_title = gets.chomp
+  app.create_rental(person_id, book_title)
+end
+
+def handle_list_rentals(app)
+  print 'Enter person ID: '
+  person_id = gets.chomp.to_i
+  app.list_rentals_by_person_id(person_id)
+end
```

Comment:
- [ Required ] As mentioned by the previous reviewer I could not check the functionality of listing rentals because I could not create a rental in the first place. Please make sure that functionality is working as well.

## Error
![Screenshot from 2023-07-01 11-59-35](https://github.com/Christelle-12/school_library/assets/16392046/04c5c7cd-8a8e-444e-b443-d4273e279637)